### PR TITLE
DDT-1134: 1.2 alignment

### PR DESCRIPTION
**Pull request by Abdiiir**

### CODE REVIEW

File: src/app/controller/services/mapping/timestamp-mapping.service.ts [hunk 1]
```diff
@@ -80,20 +80,23 @@ export class TimestampMappingService {
...
         timestamp.exportVoyageNumber = transportCall.exportVoyageNumber;
+        timestamp.carrierVoyageNumber = timestamp.carrierVoyageNumber;
         timestamp.carrierServiceCode = transportCall.carrierServiceCode;
         timestamp.transportCallSequenceNumber = transportCall.transportCallSequenceNumber;
+        if (!transportCall.exportVoyageNumber && transportCall.carrierExportVoyageNumber) {
+          // Recieve & convert to JIT 1.1 voyage numbers
+          timestamp.exportVoyageNumber = transportCall.carrierExportVoyageNumber;
+          timestamp.importVoyageNumber = transportCall.carrierImportVoyageNumber;
+        }
       }
```
Comment (nt-as) [inline src/app/controller/services/mapping/timestamp-mapping.service.ts:91]: When sending (`POST /timestamps`) do we ensure that `carrierXVoyageNumber` is set? If not, then I think it is missing.

Comment (Abdiiir) [reply src/app/controller/services/mapping/timestamp-mapping.service.ts:91]: We do that on `ensureVoyageNumbers(timestamp: Timestamp)` method, on line 34.

Comment (nt-as) [reply src/app/controller/services/mapping/timestamp-mapping.service.ts:91]: The code in `ensureVoyageNumbers` I see here does not handle ensuring that `carrierImportVoyageNumber` is set (ditto for `carrierExportVoyageNumber`)

Comment (Abdiiir) [reply src/app/controller/services/mapping/timestamp-mapping.service.ts:91]: Aaah I see what you mean. This was a design choice, the UI uses JIT 1.1 combo - newer fields are converted/downgraded to this. I added a comment that addresses this in the `transport-call.ts` TO. 

We still respect the 1.2 standard, this was mostly done to not keep constant track of both (1.1 & 1.2 voyage numbers) throughout the UI.

Do you have another approach for how we should do this?

Comment (nt-as) [reply src/app/controller/services/mapping/timestamp-mapping.service.ts:91]: The point is not which field is used internally in the UI. It is that on submit, we need to include these new fields in addition to the older field variants.

Comment (Abdiiir) [reply src/app/controller/services/mapping/timestamp-mapping.service.ts:91]: I do not see why we should, as the standard is respected & it is the responsibility of API to handle these differences. 

I will however add it  👍

Comment (Abdiiir) [reply src/app/controller/services/mapping/timestamp-mapping.service.ts:91]: Another reason we do not do that is the ref-impl does not allow mixing naming standards across versions for voyage numbers. 

We could upgrade the UI to use the JIT 1.2 voyage numbers combo, but I again do not see why we should.

Comment (nt-as) [reply src/app/controller/services/mapping/timestamp-mapping.service.ts:91]: The entire exercise is to make the UI support JIT 1.2.  In JIT 1.2 (for better or for worse), you are expected to (also) use `carrierImportVoyageNumber` and `carrierExportVoygageNumber` in addition to `importVoyageNumber` and `exportVoyageNumber`.

We can both agree that it annoying and sucks - but that is the standard and the assignment.

(As I said, I don't care what the UI uses internally, but on receive and send it needs to know about these fields)

**Review by nt-as — APPROVED** [review]

**Pull request merged**